File: app/src/main/java/com/zzq/lifeutil/adapter/GankAdapter.java
package com.zzq.lifeutil.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.paging.PagedListAdapter;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.zzq.lifeutil.R;
import com.zzq.lifeutil.net.response.FuliDataBean;
import com.zzq.lifeutil.widget.RatioImageView;

public class GankAdapter extends PagedListAdapter<FuliDataBean.ResultsBean, GankAdapter.GankViewHolder> {

    private Context mContext;

    public GankAdapter(Context context) {
        super(diffCallback);
        mContext = context;
    }

    static DiffUtil.ItemCallback<FuliDataBean.ResultsBean> diffCallback = new DiffUtil.ItemCallback<FuliDataBean.ResultsBean>() {
        @Override
        public boolean areItemsTheSame(FuliDataBean.ResultsBean oldItem, FuliDataBean.ResultsBean newItem) {
            // User properties may have changed if reloaded from the DB, but ID is fixed
            return oldItem.get_id().equals(newItem.get_id());
        }

        @Override
        public boolean areContentsTheSame(FuliDataBean.ResultsBean oldItem, FuliDataBean.ResultsBean newItem) {
            // NOTE: if you use equals, your object must properly override Object#equals()
            // Incorrectly returning false here will result in too many animations.
            return oldItem.equals(newItem);
        }
    };

    @Override
    public GankViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new GankViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_gank, parent, false));
    }

    @Override
    public void onBindViewHolder(final GankViewHolder holder, int position) {
        holder.setDataAndListener(getItem(position));
    }

    public static class GankViewHolder extends RecyclerView.ViewHolder {
        RatioImageView mImageView;
        View card;

        GankViewHolder(View view) {
            super(view);
            card = view;
            mImageView = view.findViewById(R.id.iv_item_gank);
            mImageView.setOriginalSize(50, 50);
        }

        public void setDataAndListener(final FuliDataBean.ResultsBean resultsBean) {
            Glide.with(itemView.getContext())
                    .load(resultsBean.getUrl())
                    .centerCrop()
                    .thumbnail(0.1f)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .into(mImageView);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {

                }
            });
        }
    }
}

File: PictureModule/src/main/java/com/zzq/life/picture/adapter/GankAdapter.kt
package com.zzq.life.picture.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.zzq.life.picture.R
import com.zzq.life.picture.adapter.GankAdapter.GankViewHolder
import com.zzq.life.picture.net.response.GankPictureData.DataBean

class GankAdapter(private val mContext: Context) :
    PagingDataAdapter<DataBean, GankViewHolder>(diffCallback) {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): GankViewHolder {
        return GankViewHolder(
            LayoutInflater.from(mContext)
                .inflate(R.layout.item_show_picture, parent, false)
        )
    }

    override fun onBindViewHolder(holder: GankViewHolder, position: Int) {
        holder.setDataAndListener(getItem(position))
    }

    class GankViewHolder internal constructor(view: View) :
        RecyclerView.ViewHolder(view) {
        var mImageView: ImageView
        fun setDataAndListener(resultsBean: DataBean?) {
            Glide.with(itemView.context)
                .load(resultsBean!!.url) //                    .asBitmap()
                .centerCrop()
                .thumbnail(0.1f)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(mImageView)
        }

        init {
            mImageView =
                view.findViewById(R.id.iv_item_picture)
        }
    }

    companion object {
        var diffCallback: DiffUtil.ItemCallback<DataBean> =
            object : DiffUtil.ItemCallback<DataBean>() {
                override fun areItemsTheSame(
                    oldItem: DataBean,
                    newItem: DataBean
                ): Boolean {
                    // User properties may have changed if reloaded from the DB, but ID is fixed
                    return oldItem._id == newItem._id
                }

                override fun areContentsTheSame(
                    oldItem: DataBean,
                    newItem: DataBean
                ): Boolean {
                    // NOTE: if you use equals, your object must properly override Object#equals()
                    // Incorrectly returning false here will result in too many animations.
                    return oldItem == newItem
                }
            }
    }

}